### PR TITLE
[202405][REBASE&FF]MdeModulePkg: VariableRuntimeDxe: Add host-based unit tests

### PR DESCRIPTION
**Pull request by Javagedes**
## Description

Adds hostbased unit tests for VariableRuntimeDxe including converting some Self Certification Test (SCT)s into Host Based Unit Tests.

Includes:
https://github.com/microsoft/mu_basecore/commit/c060edec1f44f2995ecd8dcfce4d5a47ec11f139
https://github.com/microsoft/mu_basecore/commit/d1ed38e06d4692123b42059185a5579bc72bfde5
https://github.com/microsoft/mu_basecore/commit/863063fd0e305e6b7b95a132f1d3e74c77ff4f98
https://github.com/microsoft/mu_basecore/commit/68670a0
https://github.com/microsoft/mu_basecore/commit/090e04916fd0eaf2ad3a3e5c2bbcbc7275c465b7

- [ ] Impacts functionality?
  - **Functionality** - Does the change ultimately impact how firmware functions?
  - Examples: Add a new library, publish a new PPI, update an algorithm, ...
- [ ] Impacts security?
  - **Security** - Does the change have a direct security impact on an application,
    flow, or firmware?
  - Examples: Crypto algorithm change, buffer overflow fix, parameter
    validation improvement, ...
- [ ] Breaking change?
  - **Breaking change** - Will anyone consuming this change experience a break
    in build or boot behavior?
  - Examples: Add a new library class, move a module to a different repo, call
    a function in a new library class in a pre-existing module, ...
- [x] Includes tests?
  - Creates multiple new tests for VariableRuntimeDxe
- [ ] Includes documentation?
  - **Documentation** - Does the change contain explicit documentation additions
    outside direct code modifications (and comments)?
  - Examples: Update readme file, add feature readme file, link to documentation
    on an a separate Web page, ...

## How This Was Tested

<_Please describe the test(s) that were run to verify the changes._>

## Integration Instructions

<_Describe how these changes should be integrated. Use N/A if nothing is required._>

### CODE REVIEW

File: MdeModulePkg/Universal/Variable/RuntimeDxe/RuntimeDxeUnitTest/BlackBoxTest/Guid.c [hunk 1]
```diff
@@ -0,0 +1,204 @@
...
+  Copyright 2006 - 2016 Unified EFI, Inc.<BR>
+  Copyright (c) 2010 - 2018, Intel Corporation. All rights reserved.<BR>
+
+  This program and the accompanying materials
+  are licensed and made available under the terms and conditions of the BSD License
+  which accompanies this distribution.  The full text of the license may be found at
+  http://opensource.org/licenses/bsd-license.php
+
+  THE PROGRAM IS DISTRIBUTED UNDER THE BSD LICENSE ON AN "AS IS" BASIS,
+  WITHOUT WARRANTIES OR REPRESENTATIONS OF ANY KIND, EITHER EXPRESS OR IMPLIED.
```
Comment (makubacki) [inline MdeModulePkg/Universal/Variable/RuntimeDxe/RuntimeDxeUnitTest/BlackBoxTest/Guid.c]: Although from SCT, as new file additions, the copyright should follow the new format with the SPDX identifier. The block comment below is also older style and could be folded into a single comment.

Comment (makubacki) [reply MdeModulePkg/Universal/Variable/RuntimeDxe/RuntimeDxeUnitTest/BlackBoxTest/Guid.c]: Similar for other files.

**Review by makubacki — APPROVED** [review]

File: MdeModulePkg/Universal/Variable/RuntimeDxe/RuntimeDxeUnitTest/BlackBoxTest/Guid.c (hunk 1 above)
Comment (Javagedes) [reply MdeModulePkg/Universal/Variable/RuntimeDxe/RuntimeDxeUnitTest/BlackBoxTest/Guid.c]: Updated all new files to the new SPDX identifier format

File: MdeModulePkg/Test/MdeModulePkgHostTest.dsc [hunk 2]
```diff
@@ -1,66 +1,126 @@
...
+      UefiSortLib|MdeModulePkg/Library/UefiSortLib/UefiSortLib.inf
+      DevicePathLib|MdePkg/Library/UefiDevicePathLib/UefiDevicePathLib.inf
+  }
+
+  MdeModulePkg/Library/ImagePropertiesRecordLib/UnitTest/ImagePropertiesRecordLibUnitTestHost.inf {
+    <LibraryClasses>
+      ImagePropertiesRecordLib|MdeModulePkg/Library/ImagePropertiesRecordLib/ImagePropertiesRecordLib.inf
+      PeCoffGetEntryPointLib|MdePkg/Library/BasePeCoffGetEntryPointLib/BasePeCoffGetEntryPointLib.inf
+  }
+
```
Comment (os-d) [inline MdeModulePkg/Test/MdeModulePkgHostTest.dsc]: Did line endings change here?

**Review by os-d — APPROVED** [review]

File: MdeModulePkg/Test/MdeModulePkgHostTest.dsc (hunk 2 above)
Comment (Javagedes) [reply MdeModulePkg/Test/MdeModulePkgHostTest.dsc]: Yes. Thanks. I fixed now :)

**Pull request merged**